src/js/providers/tracks-mixin.js

```diff
@@ -97,7 +97,8 @@ define(['../utils/underscore',
             }
         }
 
-        if (this._renderNatively) {
+        // If provider is being used to play an ad, don't add listener, so tracks won't change.
+        if (this._renderNatively && !this.instreamMode) {
```

Comment:
Checking if we're in `instreamMode` here means we may not end up removing an existing listener. It's important that we remove the listener at all times, but only add it when we're not in `instreamMode`.